pkg/builder/statefulset_container_builder.go

```diff
@@ -102,6 +103,49 @@ func (b *Builder) buildGaleraAgentContainer(mariadb *mariadbv1alpha1.MariaDB) co
 
 func buildStsInitContainers(mariadb *mariadbv1alpha1.MariaDB) []corev1.Container {
 	initContainers := []corev1.Container{}
+	if mariadb.Spec.BootstrapFrom != nil && *mariadb.Spec.BootstrapFrom.Type == "mariabackup" && mariadb.Spec.BootstrapFrom.BackupRef != nil {
+
+		cmdsCheck := []string{
+			"echo 'Checking if a restore already performed...'",
+			"[ -e /var/lib/mysql/restore_success ] && { echo 'No restore required, exiting...'; exit 0; }",
+		}
+
+		cmdsRestore := []string{
```

Comment:
I believe this is inspired in a previous @kvaps suggestion 👍🏻 :
https://github.com/mariadb-operator/mariadb-operator/pull/273#discussion_r1398243571
Would love to have some input 🙏🏻

Reply:
this is the part the init container performs a restore from mariadbbackup and once the restore is complete it creates a file to avoid another restore once the pod gets restarted.

So after every restart it checks if this file exists, if this file exists then that means the restore is already performed and just skips restore and then the actual container comes up.

```bash
restore-init Checking if a restore is already performed...                                                                                                                                                                                                     
restore-init No restore required, exiting...
```

Reply:
Yes, I think it looks good. As an alternative we could pre-populate the PVCs with a Job, but I think this is much simpler.

This would probably work when upscaling the cluster and creating new replicas, as long as the backup is in the right location.

Reply:
yes, this works, and it starts replicating from the master as long as the master has not purged the binary logs

> This would probably work when upscaling the cluster and creating new replicas, as long as the backup is in the right location.

Reply:
> purged the binary logs

exactly, we have some issues open about this:
https://github.com/mariadb-operator/mariadb-operator/issues/141